File: frontend/src/types/relation.ts
export interface Relation {
  id: string
  source_table_id: string
  source_table_name: string
  source_columns: string[]
  target_table_id: string
  target_table_name: string
  target_columns: string[]
  relation_type: 'FOREIGN_KEY' | 'INFERRED' | 'AI_SUGGESTED' | 'QUERY_INFERRED'
  confidence: number
  source: string
}

export interface RelationListResponse {
  items: Relation[]
  total: number
}

export interface QueryRelationPreview {
  temp_id: string
  source_table: string
  source_columns: string[]
  target_table: string
  target_columns: string[]
  join_type: string
  confidence: number
  already_exists: boolean
}

export interface QueryRelationResponse {
  dialect: string
  queries_parsed: number
  relations: QueryRelationPreview[]
  unmatched_tables: string[]
}

export interface SaveRelationItem {
  id: string
  source_table: string
  source_columns: string[]
  target_table: string
  target_columns: string[]
  relation_type: string
  confidence: number
}

export interface SaveRelationResponse {
  saved: number
  skipped: number
  relations: SaveRelationItem[]
}

File: backend/app/schemas/relation.py
from pydantic import BaseModel


class RelationResponse(BaseModel):
    id: str
    source_table_id: str
    source_table_name: str = ""
    source_columns: list[str]
    target_table_id: str
    target_table_name: str = ""
    target_columns: list[str]
    relation_type: str
    confidence: float
    source: str | None = None

    model_config = {"from_attributes": True}


class RelationListResponse(BaseModel):
    items: list[RelationResponse]
    total: int
